### DISCUSSION TITLE
How to get geometry given a cell ID?

### DISCUSSION THREAD

**Discussion by rmx90210** [bytesandbrains/h3-pg#124] [Q&A] [answered]
Hello, again!

I'm trying to get all hexagon IDs and geometries to cover a polygon at a given resolution. My query to get the IDs looks like:

```sql
select
	h3_polygon_to_cells(
		st_setsrid(st_extent(geom), 4326),
		8
	) as id
from boundary
```

and that's working great! Now I just need to get the geometry for each hexagon. I'm having some trouble finding the right function. I've been trying `h3_cells_to_multi_polygon_geometry()`, but that seems to return a single geometry for _all_ the hexagons, with no attributes (such as ID).

Is there anyone who might be able to give me a pointer on this? Thanks so much!

**Comment by zachasme** [comment, marked as answer]
Hi again, Robert!

If you wish to fetch one geometry per cell, I would use `h3_polygon_to_cells`, for example:

```pgsql
WITH cells AS ( 
	SELECT
		h3_polygon_to_cells(
			st_setsrid(st_extent(geom), 4326),
			8
		) AS id
	FROM boundary
) SELECT id, h3_cell_to_boundary_geometry(id) FROM cells;
```
Cheers!

**Reply by rmx90210** [reply]
Amazing, thank you @zachasme! Remembering that one 😊
